packages/catalog-process/src/services/catalogService.ts

```diff
@@ -1680,41 +1761,142 @@ export function catalogServiceBuilder(
       );
       return updatedEservice;
     },
+    async approveDelegatedEServiceDescriptor(
+      eserviceId: EServiceId,
+      descriptorId: DescriptorId,
+      { authData, correlationId, logger }: WithLogger<AppContext>
+    ): Promise<void> {
+      logger.info(`Approving EService ${eserviceId} version ${descriptorId}`);
+      const eservice = await retrieveEService(eserviceId, readModelService);
+
+      assertRequesterIsProducer(eservice.data.producerId, authData);
+
+      const descriptor = retrieveDescriptor(descriptorId, eservice);
+
+      if (descriptor.state !== descriptorState.waitingForApproval) {
+        throw notValidDescriptor(descriptor.id, descriptor.state.toString());
```

Comment:
in this case I suggest to use `notValidDescriptorState` it's more clear that this error is related on descriptor state. 
```suggestion
        throw notValidDescriptorState(descriptor.id, descriptor.state.toString());
```